src/jit/codegencommon.cpp

```diff
@@ -10622,6 +10635,19 @@ void CodeGen::genRestoreCalleeSavedFltRegs(unsigned lclFrameSize)
     // fast path return
     if (regMask == RBM_NONE)
     {
+#ifdef FEATURE_AVX_SUPPORT
+        // Before return, check if the method contains 256bit AVX instruction, issue a Vzeroupper to zero out 
+        // upper 128-bits of all YMM regs. This is to avoid AVX-256 to legacy SSE transition penalty.
+        if (compiler->getFloatingPointInstructionSet() == InstructionSet_AVX &&
+            getEmitter()->Contains256bitAVX())
+        {
+            instGen(INS_vzeroupper);
+        }
+
+        // reset both AVX and 256bit AVX flags for the function before return
+        getEmitter()->SetContainsAVX(false);
+        getEmitter()->SetContains256bitAVX(false);
+#endif
         return;
```

Comment:
Rather than duplicate this code, perhaps this could be factored out into a small method, e.g. `CodeGen::genVZeroUpperIfNeeded`.
Also, why are the flags reset before returning?